azkaban-common/src/main/java/azkaban/server/session/SessionCache.java

```diff
@@ -53,12 +56,26 @@
   public SessionCache(final Props props) {
     this.effectiveSessionTimeToLive = props.getLong(ConfigurationKeys.SESSION_TIME_TO_LIVE,
         DEFAULT_SESSION_TIME_TO_LIVE);
+
+    Long maxNumberOfSessions = null;
+    try {
+      maxNumberOfSessions = props.getLong(ConfigurationKeys.MAX_SESSION_NUMBER_PER_IP_PER_USER);
```

Comment:
Should we modify the getLong() method to return an Optional<Long>? It seems this is the only place this method is used so the refactor should be easy?

Reply:
getLong is too generic as the name suggests. Modifying it to return Optional may not be ideal. A wrapper on top of it is possible.

Reply:
I agree it's better for getLong to return null instead throwing exception when key is not found, but involves a bigger change, will make it a todo.